diff --git a/zdnn/stickify.c b/zdnn/stickify.c
index c45256b..8138369 100644
--- a/zdnn/stickify.c
+++ b/zdnn/stickify.c
@@ -41,7 +41,7 @@
 /// prefetch
 ///
 static inline void prefetch_read(const void *ptr, uintptr_t offset) {
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined(__clang__)
   __dcbt((void *)((uintptr_t)ptr + offset));
 #else
   __builtin_prefetch((void *)((uintptr_t)ptr + offset), 0);
@@ -54,7 +54,7 @@ static inline void prefetch_read(const void *ptr, uintptr_t offset) {
 /// prefetch
 ///
 static inline void prefetch_write(const void *ptr, uintptr_t offset) {
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined(__clang__)
   __dcbtst((void *)((uintptr_t)ptr + offset));
 #else
   __builtin_prefetch((void *)((uintptr_t)ptr + offset), 1);
@@ -66,7 +66,7 @@ static inline void prefetch_write(const void *ptr, uintptr_t offset) {
 /// \param[in] offset from the base address to the memory block to flush
 ///
 static inline void cache_flush(const void *ptr, uintptr_t offset) {
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined(__clang__)
   __dcbf((void *)((uintptr_t)ptr + offset));
 #else
   // No equivalent for non-MVS, leave empty or add necessary code
@@ -404,6 +404,7 @@ zdnn_status handle_fp_errors(int fe) {
                        "tuning.",
                        NO_ARG); // invalid op (bit 8) or overflow (bit 10)
   }
+#if !(defined(__MVS__) && defined(__clang__))
   if (fe & FE_INEXACT) {
     return ZDNN_STATUS(ZDNN_CONVERT_FAILURE,
                        "Internal error or Live migration happened"
@@ -411,6 +412,7 @@ zdnn_status handle_fp_errors(int fe) {
                        NO_ARG); //  inexact (bit 12)
   }
 
+#endif
   return ZDNN_STATUS_OK;
 }
 
@@ -431,6 +433,7 @@ zdnn_status handle_fp_errors_saturation(int fe, zdnn_data_types type) {
     return ZDNN_STATUS(ZDNN_ELEMENT_RANGE_VIOLATION,
                        "Range violation on tensor data", NO_ARG);
   }
+#if !(defined(__MVS__) && defined(__clang__))
   if (fe & FE_INEXACT) {
     return ZDNN_STATUS(ZDNN_CONVERT_FAILURE,
                        "Internal error or Live migration happened"
@@ -438,6 +441,7 @@ zdnn_status handle_fp_errors_saturation(int fe, zdnn_data_types type) {
                        NO_ARG); //  inexact (bit 12)
   }
 
+#endif
   return ZDNN_STATUS_OK;
 }
 /// Call HW to transform from FP32 -> DLFLOAT16 only.
