diff --git a/zdnn/zdnn.c b/zdnn/zdnn.c
index ca705ea..4228859 100644
--- a/zdnn/zdnn.c
+++ b/zdnn/zdnn.c
@@ -23,7 +23,7 @@
 #include <string.h>
 #include <strings.h>
 
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined(__clang__)
 #include <cvt.h>
 #include <ihaecvt.h>
 #include <ihafacl.h>
@@ -276,7 +276,7 @@ zdnn_status invoke_nnpa(uint8_t function_code, char *parm_block,
   }
 
   // clang-format off
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined(__clang__)
   struct psa *psaptr =
       (struct psa *)0;
   // set _cvt to x10, the pointer to the CVT on z/OS
@@ -305,21 +305,23 @@ zdnn_status invoke_nnpa(uint8_t function_code, char *parm_block,
         return ZDNN_STATUS(ZDNN_UNAVAILABLE_FUNCTION,
                        "NNPA facility unavailable", NO_ARG);
   }
-
-#else
+#elif defined(__clang__) && defined(__MVS__)
+  // clang on z/OS: use STFLE to check NNPA, use .long encoding like Linux
+  if (zdnn_is_nnpa_installed()) {
     register uint64_t r0 __asm__("%r0") = function_code;
-    register uint64_t r1 __asm__("%r1") = (uint64_t)parm_block;
-
-    __asm__ __volatile__ (
-          "1: .long   0xb93b0000"     "\n\t"   // NNPA
-          "   jo      1b"             "\n\t"   // on CC=3, jump to label '1'
-          "   ipm     %[cc]"          "\n\t"   // fetch cc to cc reg
-          "   srl     %[cc],28"       "\n\t"   // shift
-    : [r0] "+d" (r0), [cc] "+d" (cc)          // ASM outputs
-    : "d" (r1)                           // ASM inputs
-    : "memory", "cc");                   // ASM clobbers
+    register uint64_t r1 __asm__("%r1") = (uint64_t)(uintptr_t)parm_block;
+    __asm volatile(
+         "         DC X'B93B0000'             \n"
+         "         JO *-4                     \n"
+         "         IPM   %[cc]               \n"
+         "         SRL   %[cc],28            \n"
+         : [cc] "+d" (cc), "+d" (r0)
+         : "d" (r1)
+         : "memory", "cc");
     rtn.r0 = r0;
-
+  } else {
+    return ZDNN_STATUS(ZDNN_UNAVAILABLE_FUNCTION, "NNPA facility unavailable", NO_ARG);
+  }
 #endif   // defined(__MVS__)
   // clang-format on
 
@@ -353,7 +355,7 @@ zdnn_status invoke_nnpa(uint8_t function_code, char *parm_block,
 /// possible on NNPA_QAF.
 ///
 zdnn_status invoke_nnpa_query(nnpa_qaf_parameter_block *qpb) {
-#if defined(__MVS__)
+#if defined(__MVS__) && !defined (__clang__)
   /***********************************************************************
    * On z/OS, use system copy of STFLE output ("faclnnpaf").  (LoZ has to
    * worry about dynamic changes to STFLE.  z/OS does not support that so
